### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by IdanTo1**

### CODE REVIEW

File: src/main/java/bgu/spl/mics/MessageBrokerImpl.java [hunk 1]
```diff
@@ -80,10 +84,13 @@ public void sendBroadcast(Broadcast b) {
 	 * @post all subscribers subscribed to e.getClass() queue's last element will be e.
 	 */
 	public <T> Future<T> sendEvent(Event<T> e) {
-		Subscriber currentSub = _eventSubscribers.get(e.getClass()).poll();
+		Subscriber currentSub = null;
+		synchronized (_eventSubscribers.get(e.getClass())) {
+			currentSub = _eventSubscribers.get(e.getClass()).poll();
+			_eventSubscribers.get(e.getClass()).add(currentSub);
```
Comment (IdanTo1) [inline src/main/java/bgu/spl/mics/MessageBrokerImpl.java]: _eventSubscribers.get(e.getClass()) is called 3 times.. maybe parameter?

**Review by Hroy02 — DISMISSED** [review]
Looks great!

**Review by IdanTo1 — COMMENTED** [review]
looks great, only one change for your choice

**Review by Hroy02 — APPROVED** [review]
Great!

**Pull request merged**